lib/phenotype/versioned_app.rb

```diff
@@ -0,0 +1,48 @@
+module Phenotype
+  class VersionedApp
+    attr_accessor :strategies, :routes
+    attr_writer :no_strategy_response, :too_many_strategies_response, :version_not_found_response
+
+    def initialize(strategies: [])
+      @routes = {}
+      @strategies = strategies
+    end
+
+    def add_strategy(strat)
+      strategies.push(strat)
+    end
+
+    def add_version(version:, &block)
+      routes[version] = block
+    end
+
+    def call(env)
+      strats = strategies.find_all { |strat| strat.version(env) }
+      return no_strategy_response.call(env) if strategies.empty?
```

Comment:
I just realized, how are we going to handle versioning with apps where we can't necessarily change the client namely mobile API and old versions of our mobile apps?